// Copyright 2015 The gRPC Authors
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
syntax = "proto3";

option java_multiple_files = true;
option java_package = "seqprediction";
option java_outer_classname = "SeqPredictionProto";
option objc_class_prefix = "SQP";

package seqprediction;

// The sequence prediction service definition.
service Predictor {
  rpc setNumTrees (SetNumTreesMessage) returns (SetNumTreesMessage) {}
  rpc predict (SequenceMessage) returns (SequenceMessage) {}
  rpc train (SequenceMessage) returns (TrainResponse) {}
}

message SequenceMessage {
    int32 seqId = 1;
    int32 treeId = 2;
    repeated int32 seq = 3;
    double runtimeInSeconds = 4;
}

message TrainResponse {
    int32 treeId = 1;
    bool result = 2;
    double runtimeInSeconds = 3;
}

message SetNumTreesMessage {
    int32 numTrees = 1;
    bool result = 2;
}
